src/status/libstatus/contacts.nim

```diff
@@ -1,45 +1,34 @@
-import json, strmisc, atomics
+import json, strmisc
 import core, ../utils, ../types
-from ../profile/profile import Profile
 
-var
-  contacts {.threadvar.}: JsonNode
-  contactsInited {.threadvar.}: bool
-  dirty: Atomic[bool]
+# this module is made stateless intentionally
+# all caching logic is done in status/contacts.nim
 
-# TODO: remove Profile from here
-proc blockContact*(contact: Profile): string =
+proc blockContact*(id: string, ensVerified: bool, ensName: string, alias: string, identicon: string, systemTags: seq[string], localNickname: string): string =
```

Comment:
Why did you change this to pass all individually instead? I don't think it should make anyway performance gain and we call this function using a Profile anyway.